feat: organization upgrade flow v3

sean-brydon opened the pull request:
Video: https://www.loom.com/share/28d64ee3c5b940698b2127f63332da1f

## What does this PR do?

Adds an organization onboarding v3 migration flow that lets users move existing teams and their members into a new organization. Introduces new steps, UI, store fields, and submission/redirect logic tailored for migrations.

## How to test

1. Turn on `onboarding-v3` feature flag
2. Test with Stripe disabled + Stripe enabled
3. Go to http://app.cal.local:3000/settings/my-account/profile and click upgrade
4. Test migration flow with existing teams

## Updates since last revision

Addressed PR feedback from Udit and Eunjae:
- Use CSS `uppercase` class instead of `toUpperCase()` for team name initials
- Use `border-default` instead of hardcoded `border-white` for brand colors
- Use early return pattern to avoid nested ifs in useEffect
- Disable continue button while migration flow is loading (`isLoading` from `useMigrationFlow`)
- Preserve team member roles (OWNER→ADMIN, MEMBER, ADMIN) instead of hardcoding MEMBER
- Use `findIfAMemberOfSomeOrganization` for efficient existence check in migrate-members page
- Fixed test to include `role` field in `migratedMembers` mock data to match updated store type

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a documentation change. N/A
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## Checklist for human review

- [ ] Verify the OWNER → ADMIN role conversion is the correct business logic for migrated team members
- [ ] Verify `findIfAMemberOfSomeOrganization` is semantically equivalent to checking `organizations.length > 0`
- [ ] Test the continue button disabled state during loading doesn't cause UX issues

---

**Link to Devin run:** https://app.devin.ai/sessions/005ee5c0dbaf4702b1fd0b1cd27aba0c
**Requested by:** @sean-brydon (sean@cal.com)

sean-brydon committed:
wip flow

sean-brydon committed:
add tests

sean-brydon committed:
WIP migrateing view

sean-brydon committed:
push back step

sean-brydon committed:
fix tests and logic for adding new members to existing teams

sean-brydon committed:
few UI fixes

sean-brydon committed:
type fixes

sean-brydon committed:
fix nits

sean-brydon committed:
few UI + re-route fixes

sean-brydon committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

sean-brydon committed:
fix teamId when migrating

eunjae-lee commented on apps/web/modules/onboarding/organization/migrate-teams/organization-migrate-teams-view.tsx:
```diff
@@ -0,0 +1,234 @@
+  const moveTeams = watch("moveTeams");
+
+  useEffect(() => {
+    if (teams.length > 0) {
```
nit: maybe early return to avoid nested ifs

eunjae-lee reviewed (dismissed):
some nits but not blocking. working well.

sean-brydon committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

sean-brydon committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

keithwillcode committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

JagjeevanAK reviewed (commented):
LGTM

CarinaWolli commented:
@sean-brydon can you fix the type errors?

sean-brydon committed:
fix types

sean-brydon committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

devin-ai-integration[bot] committed:
fix: address PR feedback from Udit and Eunjae

- Use CSS uppercase class instead of toUpperCase() for team name initials
- Use border-default instead of hardcoded border-white for brand colors
- Use early return pattern to avoid nested ifs in useEffect
- Disable continue button while migration flow is loading
- Preserve team member roles (OWNER->ADMIN, MEMBER, ADMIN) instead of hardcoding MEMBER
- Use findIfAMemberOfSomeOrganization for efficient existence check

Co-Authored-By: sean@cal.com <Sean@brydon.io>

devin-ai-integration[bot] committed:
fix: update test to include role field in migratedMembers mock data

Co-Authored-By: sean@cal.com <Sean@brydon.io>

devin-ai-integration[bot] committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

Resolved merge conflict in apps/web/modules/onboarding/components/onboarding-invite-browser-view.tsx:
- Preserved PR's migration flow features (migratedMembers, teams store fields)
- Preserved PR's useOrganizationInvites conditional logic for banner display
- Incorporated main's structural improvements (classNames utility, null fallback)
- Combined both changes to maintain functionality from both branches

Co-Authored-By: unknown <>

devin-ai-integration[bot] committed:
Merge branch 'main' into feat/orgs-migrate-to-onboarding-v3

Resolved merge conflict in apps/web/modules/onboarding/components/onboarding-invite-browser-view.tsx:
- Preserved PR's new functionality for migratedMembers and teams from useOnboardingStore
- Kept PR's enhanced banner/avatar display logic for useOrganizationInvites
- Maintained classNames import for conditional styling
- Applied main's formatting improvements where compatible

Co-Authored-By: unknown <>
